CIME/case/case_submit.py

```diff
@@ -297,6 +293,54 @@ def check_case(self, skip_pnl=False, chksum=False):
         # the ww3 buildnml has dependencies on inputdata so we must run it again
         self.create_namelists(component="WAV")
 
+    if self.get_value("COMP_INTERFACE") == "nuopc":
+        #
+        # Check that run length is a multiple of the longest component
+        # coupling interval. The longest interval is smallest NCPL value.
+        # models using the nuopc interface will fail at initialization unless
+        # ncpl follows these rules, other models will only fail later and so
+        # this test is skipped so that short tests can be run without adjusting NCPL
+        #
+        maxncpl = 10000
+        minncpl = 0
+        maxcomp = None
+        for comp in self.get_values("COMP_CLASSES"):
+            if comp == "CPL":
+                continue
+            ncpl = self.get_value("{}_NCPL".format(comp))
+            if ncpl and maxncpl > ncpl:
+                maxncpl = ncpl
+                maxcomp = comp
+            if ncpl and minncpl < ncpl:
+                minncpl = ncpl
+
+        ncpl_base_period = self.get_value("NCPL_BASE_PERIOD")
+        if ncpl_base_period == "hour":
+            coupling_secs = 3600 / maxncpl
+            timestep = 3600 / minncpl
+        elif ncpl_base_period == "day":
+            coupling_secs = 86400 / maxncpl
+            timestep = 86400 / minncpl
+        elif ncpl_base_period == "year":
+            coupling_secs = 3.154e7 / maxncpl
+            timestep = 3.154e7 / minncpl
+        elif ncpl_base_period == "decade":
+            coupling_secs = 3.154e8 / maxncpl
+            timestep = 3.154e8 / minncpl
```

Comment:
Is this going to work robustly given that you're using floats rather than ints? I wonder if you should change this to either use ints or round the result to the nearest int???

Reply:
These values are not being used for anything other than this message so I don't thing it matters if they are real or integer.

Reply:
They're used below in `runtime % coupling_secs`. I just tested this and I think it would give a problem currently for ncpl_base_period == "year" or ncpl_base_period == "decade" - e.g.:

```python
maxncpl = 365
runtime = 86400
coupling_secs = 3.154e7 / maxncpl
runtime % coupling_secs
```

gives 86400.0.

Whereas this (corrected):

```python
maxncpl = 365
runtime = 86400
coupling_secs = 31536000 / maxncpl
runtime % coupling_secs
```

gives 0.0, correctly.